src/gui/mesh/qgsmeshdatasetgrouptreeview.cpp

```diff
@@ -494,17 +494,25 @@ void QgsMeshActiveDatasetGroupTreeView::setActiveGroup()
   setActiveVectorGroup( vectorGroup );
 }
 
+QgsMeshDatasetGroupListModel::QgsMeshDatasetGroupListModel( QObject *parent ): QAbstractListModel( parent )
+{}
+
 void QgsMeshDatasetGroupListModel::syncToLayer( QgsMeshLayer *layer )
 {
+  beginResetModel();
   if ( layer )
     mRootItem = layer->datasetGroupTreeRootItem();
+  endResetModel();
 }
 
 int QgsMeshDatasetGroupListModel::rowCount( const QModelIndex &parent ) const
 {
   Q_UNUSED( parent );
   if ( mRootItem )
-    return mRootItem->totalChildCount();
+  {
+    QList<int> list = mRootItem->enabledDatasetGroupIndexes();
```

Comment:
remove unused?

Reply:
it is really unused

Reply:
so you can remove the line?

Reply:
... sorry, I was thinking you talked about the Q_UNUSED... so, of course, removed...